### DISCUSSION TITLE
Rust binary build in yocto is not running in ARM target

### DISCUSSION THREAD

**Discussion by gokulsvg** [meta-rust/meta-rust#372] [Q&A]
hello-world example in meta-rust was complied successfully , but on executing it in BMC is giving the flowing error

"Illegal instruction (core dumped)"

The same code cross compiled with a arm rust cross compiler is creating a bigger binary and its executing fine in target

**Comment by kraj** [comment]
whats your target arm subarchitecture? it seems there is a mismatch what rust compiler thinks about your target.

**Reply by gokulsvg** [reply]
The target is ARMv7, i used readelf command to get the --arch-specific details and i can see some mis match
like yocto build is having 2 extra option 
**Tag_Advanced_SIMD_arch: NEONv1
Tag_ABI_align_preserved: 8-byte, except leaf SP**

This is not present in the binary build with a rust armv7 cross compiler

 & one more difference observed is 
**Tag_FP_arch: VFPv4 [for yocto build ]
Tag_FP_arch: VFPv3-D16 [for binary compiled with armv7 cros compiler ]**

binary build with rust arm cross compiler is build with following shared objects 
 0x00000001 (NEEDED)                     Shared library: [libgcc_s.so.1]
 0x00000001 (NEEDED)                     Shared library: [librt.so.1]
 0x00000001 (NEEDED)                     Shared library: [libpthread.so.0]
 0x00000001 (NEEDED)                     Shared library: [libm.so.6]
 0x00000001 (NEEDED)                     Shared library: [libdl.so.2]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
 0x00000001 (NEEDED)                     Shared library: [ld-linux-armhf.so.3]

and for the same code for yocto build only few shared objects are used
 0x00000001 (NEEDED)                     Shared library: [libgcc_s.so.1]
 0x00000001 (NEEDED)                     Shared library: [libc.so.6]
 0x00000001 (NEEDED)                     Shared library: [ld-linux-armhf.so.3]
